test/dotcom_web/components/system_status/subway_status_test.exs

```diff
@@ -9,13 +9,167 @@ defmodule DotcomWeb.Components.SystemStatus.SubwayStatusTest do
   alias Dotcom.SystemStatus.Subway
   alias Test.Support.Factories
 
+  @lines_without_branches List.delete(Subway.lines(), "Green")
+
   setup :verify_on_exit!
 
   setup do
     stub_with(Dotcom.Utils.DateTime.Mock, Dotcom.Utils.DateTime)
     :ok
   end
 
+  describe "homepage_subway_status/1" do
+    test "renders all rows as 'Normal Service' when there are no alerts" do
+      # Setup
+      alerts = []
+
+      # Exercise
+      rows = status_rows_for_alerts(alerts)
+
+      # Verify
+      assert rows
+             |> Enum.each(fn row ->
+               assert status_label_text_for_row(row) == "Normal Service"
+             end)
+    end
+
+    test "shows all of the route pills by default" do
+      # Setup
+      alerts = []
+
+      # Exercise
+      rows = status_rows_for_alerts(alerts)
+
+      # Verify
+      assert rows
+             |> Enum.each(fn row ->
+               assert route_pill_visibility_for_row(row) == :visible
+             end)
+    end
+
+    test "hides the second route pill for a line if that line has more than one alert" do
+      # Setup
+      affected_line = Faker.Util.pick(@lines_without_branches)
+
+      effects =
+        Faker.Util.sample_uniq(2, fn -> Faker.Util.pick(service_impacting_effects()) end)
+
+      alerts =
+        effects
+        |> Enum.map(
+          &(Factories.Alerts.Alert.build(:alert_for_route, route_id: affected_line, effect: &1)
+            |> Factories.Alerts.Alert.active_during(Timex.now()))
+        )
+
+      # Exercise
+      rows = status_rows_for_alerts(alerts)
+
+      # Verify
+      assert rows
+             |> for_route(affected_line)
+             |> Enum.map(&route_pill_visibility_for_row/1) ==
+               [:visible, :invisible]
+    end
+
+    test "collapses alerts for non-branched lines if there would otherwise be too many rows" do
```

Comment:
What's the threshold for "too many"?

Reply:
[Five, obviously](https://github.com/mbta/dotcom/pull/2404/commits/ddd51adfffb8f70d6b70a2fd77b09ad374bfb34d)

😛